### ISSUE TITLE
TWAP: Test cases for LastSpotPrice Error

### ISSUE THREAD

**Issue by ValarDragon** [osmosis-labs/osmosis#2562]
## Background

There are a few test cases missing though, namely:
- [ ] That the behavior of the public API for `GetArithmeticTwap` works as expected, across potential values for the LastSpotPriceErr.

**Comment by p0mvn** [comment]
@ValarDragon I'm starting to work on this. Just FYI to avoid working on the same thing

**Comment by p0mvn** [comment]
I haven't made much progress on this yet since my other issues got unblocked. If anyone is has bandwidth to pick this up, please let me know

**Comment by p0mvn** [comment]
I split this into subcomponents. I've done some progress on #2659 and will continue working on that
